routing/datasource.go

```diff
@@ -483,6 +483,12 @@ func processTreePredicates(r *Route, predicateList []*eskip.Predicate) error {
 	return nil
 }
 
+// ProcessRouteDef processes a route definition for the routing table.
+// This function is exported to be used by validation webhooks.
+func ProcessRouteDef(o *Options, cpm map[string]PredicateSpec, def *eskip.Route) (*Route, error) {
+	return processRouteDef(o, cpm, def)
+}
+
```

Comment:
This exports internal function which is not good and will block us from refactoring it later.
I think we should provide a dedicated validation function like
```go
func ValidateRoute(o *Options, r *eskip.Route) error 
```